VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,415 @@ Function Install-EsxiCertificate {
+                "MW", "MX", "MY", "MZ", "NA", "NC", "NE", "NF", "NG", "NI", "NL", "NO", "NP", "NR", "NT", "NU", "NZ", "OM", "PA", "PE", "PF", "PG", "PH", "PK", "PL", "PM", `
+                "PN", "PR", "PS", "PT", "PW", "PY", "QA", "RE", "RO", "RS", "RU", "RW", "SA", "SB", "SC", "SE", "SG", "SH", "SI", "SJ", "SK", "SL", "SM", "SN", "SR", "ST", `
+                "SU", "SV", "SZ", "TC", "TD", "TF", "TG", "TH", "TJ", "TK", "TM", "TN", "TO", "TP", "TR", "TT", "TV", "TW", "TZ", "UA", "UG", "UM", "US", "UY", "UZ", "VA", `
+                "VC", "VE", "VG", "VI", "VN", "VU", "WF", "WS", "YE", "YT", "ZA", "ZM", "COM", "EDU", "GOV", "INT", "MIL", "NET", "ORG", "ARPA")] [String] $country,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $locality,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $organization,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $organizationUnit,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $stateOrProvince,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $email
+    )
+
+    if (Test-VCFConnection -server $server) {
+        if (Test-VCFAuthentication -server $server -user $user -pass $pass) {
+            $domainType = Get-VCFWorkloadDomain -name $workloadDomain
+            $resourcesObject = gatherSddcInventory -domainType $domainType.type -workloadDomain $workloadDomain
+            Write-Output "Populating $($workloadDomain)-requestCsrSpec.json"
+            $path = Join-Path $pwd ""
+            $csrGenerationSpecJson =
+            '{
+            "csrGenerationSpec": {
+                "country": "'+ $country + '",
+                "email": "'+ $email + '",
+                "keyAlgorithm": "'+ "RSA" + '",
+                "keySize": "'+ "2048" + '",
+                "locality": "'+ $locality + '",
+                "organization": "'+ $organization + '",
+                "organizationUnit": "'+ $organizationUnit + '",
+                "state": "'+ $stateOrProvince + '"
+                },
+            '
+            $resourcesBodyObject += [pscustomobject]@{
+                resources = $resourcesObject
+            }
+            $resourcesBodyObject | ConvertTo-Json -Depth 10 | Out-File -FilePath $path"temp.json"
+            Get-Content $path"temp.json" | Select-Object -Skip 1 | Set-Content $path"temp1.json"
+            $resouresJson = Get-Content $path"temp1.json" -Raw
+            Remove-Item -Path $path"temp.json"
+            Remove-Item -Path $path"temp1.json"
+            $requestCsrSpecJson = $csrGenerationSpecJson + $resouresJson
+            $requestCsrSpecJson | Out-File $path"$($workloadDomain)-requestCsrSpec.json"
+            Write-Output "Requesting CSRs for Components Associated with Domain $($workloadDomain)"
```

Comment:
```suggestion
            Write-Output "Requesting certificate signing requests for components associated with workload domain ($($workloadDomain))..."
```

Reply:
addressed in (131acd4bec5c68a803eb55ef8c6b8d2638b81d17)